src/python/pants/backend/project_info/paths.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import json
+from collections import deque
+from operator import attrgetter
+from typing import Dict, Generator, List, Optional, cast
```

Comment:
fyi you can use `from __future__ import annotations` to be able to write type hints like `list[str] | None` instead of `Optional[List[str]]`. We prefer this style

Reply:
Fixed